CRM/Member/Form/MembershipRenewal.php

```diff
@@ -339,6 +339,12 @@ public function buildQuickForm() {
 
     $this->applyFilter('__ALL__', 'trim');
 
+    // CRM-20571
+    // Get the Join Date from Membership info as it is not available in the Renew form
+    $joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id, 'join_date');
+    // Save its value in a hidden field
+    $this->addElement('hidden', 'join_date', $joinDate);
```

Comment:
@jitendrapurohit I will make the change you suggested, will check out some examples, let me try. Your explanation helped to understand how `$self` appeared. I edited my reply to give you the link to my query on Stackexchange - I wanted to know how to access `$this->_id` in `formFule()`. If this work in `formRule()` if we pass `$this` as you suggested then we have a better solution right away. This should work:
```
$joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $self->_id, 'join_date')
```